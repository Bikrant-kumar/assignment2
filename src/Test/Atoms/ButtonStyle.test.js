import React from 'react';
import { screen, fireEvent, render } from '@testing-library/react';
import Button from '../../Component/Atoms/ButtonStyle';
import renderer from 'react-test-renderer';

const clickData = jest.fn();

describe(Button ,() => {

  test('Button click',() => {
    render(<Button 
    clickData={clickData}
    text='submit'
    />);
   fireEvent.click(screen.getAllByRole('button')[0]);
   expect(clickData).toHaveBeenCalledTimes(1);
 })

 test('test Button with snapshot',()=>{
  const component= renderer.create(<Button clickData={clickData}
    text='submit'/>).toJSON();
  expect(component).toMatchSnapshot();
 })

})
