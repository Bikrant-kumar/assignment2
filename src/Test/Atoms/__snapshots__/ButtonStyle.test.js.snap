// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyle test Button with snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Class {
            "_dispatchInstances": null,
            "_dispatchListeners": null,
            "_targetInst": null,
            "altKey": null,
            "bubbles": null,
            "button": null,
            "buttons": null,
            "cancelable": null,
            "clientX": null,
            "clientY": null,
            "ctrlKey": null,
            "currentTarget": [Function],
            "defaultPrevented": null,
            "detail": null,
            "dispatchConfig": null,
            "eventPhase": null,
            "getModifierState": [Function],
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": null,
            "metaKey": null,
            "movementX": [Function],
            "movementY": [Function],
            "nativeEvent": null,
            "pageX": null,
            "pageY": null,
            "relatedTarget": [Function],
            "screenX": null,
            "screenY": null,
            "shiftKey": null,
            "target": null,
            "timeStamp": [Function],
            "type": null,
            "view": null,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    submit
  </span>
</button>
`;
